adal/src/telemetry/java/com/microsoft/aad/adal/CacheEvent.java

```diff
@@ -84,7 +90,7 @@ public void processEvent(final Map<String, String> dispatchMap) {
             final String name = eventPair.first;
 
             if (name.equals(EventStrings.TOKEN_TYPE_IS_FRT) || name.equals(EventStrings.TOKEN_TYPE_IS_RT)
-                    || name.equals(EventStrings.TOKEN_TYPE_IS_MRRT)) {
+                    || name.equals(EventStrings.TOKEN_TYPE_IS_MRRT) || name.equals(EventStrings.SPE_INFO)) {
```

Comment:
what's the behavior if both HttpEvent and CacheEvent have this set but with different values? e.g first CacheEvent happens and there's no SPE ring info, then token refresh is done and we get SPE ring info.

Reply:
If the stored token does not contain any spe ring info it will not appear in the telemetry. Once the refresh request happens the `HttpEvent` will report the updated spe ring data. When this value is later retrieved the `CacheEvent` will reflect the new spe ring info stored in the cache.